test/IntegrationTests/MSTest.Acceptance.IntegrationTests/TestDiscoveryTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests;
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests.Helpers;
+using Microsoft.Testing.Platform.Helpers;
+
+namespace MSTest.Acceptance.IntegrationTests;
+
+[TestGroup]
+public class TestDiscoveryTests : AcceptanceTestBase
+{
+    private readonly TestAssetFixture _testAssetFixture;
+    private const string AssetName = "TestDiscovery";
+
+    // There's a bug in TAFX where we need to use it at least one time somewhere to use it inside the fixture self (AcceptanceFixture).
+    public TestDiscoveryTests(ITestExecutionContext testExecutionContext, TestAssetFixture testAssetFixture,
+        AcceptanceFixture globalFixture)
+        : base(testExecutionContext)
+    {
+        _testAssetFixture = testAssetFixture;
+    }
+
+    [ArgumentsProvider(nameof(TargetFrameworks.All), typeof(TargetFrameworks))]
+    public async Task DiscoverTests_FindsAllTests(string currentTfm)
+    {
+        var testHost = TestHost.LocateFrom(_testAssetFixture.TargetAssetPath, AssetName, currentTfm);
+
+        var testHostResult = await testHost.ExecuteAsync("--list-tests");
+
+        testHostResult.AssertExitCodeIs(ExitCodes.Success);
+        testHostResult.AssertOutputContains($"Test1");
+        testHostResult.AssertOutputContains($"Test2");
```

Comment:
Nit: no need for the interpolated string

```suggestion
        testHostResult.AssertOutputContains("Test1");
        testHostResult.AssertOutputContains("Test2");
```